src/shapes/object.class.ts

```diff
@@ -1687,52 +1688,54 @@ import { enlivenObjectEnlivables } from '../util/misc/objectEnlive';
        * @return {Object} center point from element coordinates
        */
       _findCenterFromElement: function () {
-        return { x: this.left + this.width / 2, y: this.top + this.height / 2 };
-      },
-
-      /**
-       * This function is an helper for svg import. it decompose the transformMatrix
-       * and assign properties to object.
-       * untransformed coordinates
-       * @private
-       * @chainable
-       */
-      _assignTransformMatrixProps: function () {
-        if (this.transformMatrix) {
-          var options = fabric.util.qrDecompose(this.transformMatrix);
-          this.flipX = false;
-          this.flipY = false;
-          this.set('scaleX', options.scaleX);
-          this.set('scaleY', options.scaleY);
-          this.angle = options.angle;
-          this.skewX = options.skewX;
-          this.skewY = 0;
-        }
+        return new Point(
+          this.left + this.width / 2,
+          this.top + this.height / 2
+        );
       },
 
       /**
        * This function is an helper for svg import. it removes the transform matrix
-       * and set to object properties that fabricjs can handle
+       * and set to object properties that fabric can handle
        * @private
        * @param {Object} preserveAspectRatioOptions
        * @return {thisArg}
        */
       _removeTransformMatrix: function (preserveAspectRatioOptions) {
-        var center = this._findCenterFromElement();
-        if (this.transformMatrix) {
-          this._assignTransformMatrixProps();
-          center = fabric.util.transformPoint(center, this.transformMatrix);
+        let center = this._findCenterFromElement();
+        const t = this.transformMatrix;
+        if (t) {
+          const { scaleX, scaleY, angle, skewX } = qrDecompose(t);
+          this.flipX = false;
+          this.flipY = false;
+          this.set('scaleX', scaleX);
+          this.set('scaleY', scaleY);
+          this.angle = angle;
+          this.skewX = skewX;
```

Comment:
this is a case that you warned from
I tried calling `set` instead of manual assignment but since polys run a side effect this breaks.
I can set `initialized=false` so it won't run.

Reply:
intialized false has to go away, do not cement it.